ProjectLighthouse.Servers.Website/Pages/UserPage.cshtml.cs

```diff
@@ -18,44 +19,81 @@ public class UserPage : BaseLayout
 
     public bool CommentsEnabled;
 
-    public bool IsProfileUserHearted;
+    public List<SlotEntity>? HeartedSlots;
 
     public bool IsProfileUserBlocked;
 
+    public bool IsProfileUserHearted;
+
     public List<PhotoEntity>? Photos;
-    public List<SlotEntity>? Slots;
 
-    public List<SlotEntity>? HeartedSlots;
+    public UserEntity? ProfileUser;
     public List<SlotEntity>? QueuedSlots;
+    public List<SlotEntity>? Slots;
+
+    public bool ProfilePrivate;
+    public bool SlotsPrivate;
+    
+    public bool CommentsDisabledByModerator;
 
-    public UserEntity? ProfileUser;
     public UserPage(DatabaseContext database) : base(database)
-    {}
+    { }
 
     public async Task<IActionResult> OnGet([FromRoute] int userId)
     {
         this.ProfileUser = await this.Database.Users.FirstOrDefaultAsync(u => u.UserId == userId);
         if (this.ProfileUser == null) return this.NotFound();
 
         // Determine if user can view profile according to profileUser's privacy settings
-        if (this.User == null || !this.User.IsAdmin)
+        if (this.User == null || !this.User.IsModerator)
         {
             switch (this.ProfileUser.ProfileVisibility)
             {
+                case PrivacyType.Game:
+                {
+                    if (this.ProfileUser != this.User) this.ProfilePrivate = true;
+                    break;
+                }
                 case PrivacyType.PSN:
                 {
-                    if (this.User != null) return this.NotFound();
-
+                    if (this.User == null) this.ProfilePrivate = true;
                     break;
                 }
-                case PrivacyType.Game:
+                case PrivacyType.All:
                 {
-                    if (this.ProfileUser != this.User) return this.NotFound();
+                    this.ProfilePrivate = false;
+                    break;
+                }
+                default:
+                {
+                    this.ProfilePrivate = false;
+                    break;
+                }
+            }
 
+            // Determine if user can view levels according to profileUser's privacy settings
+            switch (this.ProfileUser.LevelVisibility)
+            {
+                case PrivacyType.Game:
+                {
+                    if (this.ProfileUser != this.User) this.SlotsPrivate = true;
+                    break;
+                }
+                case PrivacyType.PSN:
+                {
+                    if (this.User == null) this.SlotsPrivate = true;
+                    break;
+                }
+                case PrivacyType.All:
+                {
+                    this.SlotsPrivate = false;
+                    break;
+                }
+                default:
+                {
+                    this.SlotsPrivate = false;
                     break;
                 }
```

Comment:
Simplify
```suggestion
                default:
                {
                    this.SlotsPrivate = false;
                    break;
                }
```
Or if rider complains put both labels above the same block